controllers/rhmi/rhmi_controller.go

```diff
@@ -338,26 +338,26 @@ func (r *RHMIReconciler) Reconcile(ctx context.Context, request ctrl.Request) (c
 
 	installType, err := TypeFactory(ctx, installation.Spec.Type, r.Client)
 	if err != nil {
-		return ctrl.Result{}, err
+		return retryRequeue, nil
```

Comment:
I'm curious as to why we are ignoring to return `err` in some places vs not in other places 🤔 

If we dont return the error, I believe this won't be logged by the reconciler which may a valid error to know where in the code it is failing for debug purposes (For example in this instance, we may fail to determine the platform type or an unknown installation type was used - https://github.com/integr8ly/integreatly-operator/blob/master/controllers/rhmi/types.go#L131-L160)

Reply:
The decision to have the controller retriggered by an error in certain cases was made in order to promptly address important issues. However, in some cases where we are waiting for example for a resource, I’d to give it some time and requeue the controller. 

Agree, in this case with TypeFactory it makes sense to return an error. I'll review the code for any other missed error handling.

Reply:
Updated 😌